File: backend/resource_server/src/main/java/com/example/resource/entities/Part.java
package com.example.resource.entities;

import javax.persistence.*;

@Entity
@Table(name = "part")
public class Part {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name = "part_name",nullable = false)
    private String partName;

    @Column(name = "warehouse_number",nullable = false)
    private String warehouseNumber;

    public Part() {
    }

    public Part(String partName, String warehouseNumber) {
        this.partName = partName;
        this.warehouseNumber = warehouseNumber;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getPartName() {
        return partName;
    }

    public void setPartName(String partName) {
        this.partName = partName;
    }

    public String getWarehouseNumber() {
        return warehouseNumber;
    }

    public void setWarehouseNumber(String warehouseNumber) {
        this.warehouseNumber = warehouseNumber;
    }
}

File: frontend/website/src/app/models/Part.ts
export class Part {
  private _id:number;
  private _partName:string;
  private _warehouseNumber:string;
  private _highlightedPartName:string;

  constructor() {
  }


  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get partName(): string {
    return this._partName;
  }

  set partName(value: string) {
    this._partName = value;
  }

  get warehouseNumber(): string {
    return this._warehouseNumber;
  }

  set warehouseNumber(value: string) {
    this._warehouseNumber = value;
  }

  get highlightedPartName(): string {
    return this._highlightedPartName;
  }

  set highlightedPartName(value: string) {
    this._highlightedPartName = value;
  }
}
